src/main/java/com/boot/product/controller/ProductController.java

```diff
@@ -84,6 +84,16 @@ public ResponseEntity<ProductDTO> findProductBySlug(@Size(min = 3, max = 30, mes
         return new ResponseEntity<>(product, HttpStatus.OK);
     }
 
+    @GetMapping("/partialName/{partialName}")
```

Comment:
I wouldn't put the search string as a pathVariable but as a query param.
I see that the emag call looks like:
/search-suggester?source_id=7&query=masina&shop_id=1